server/util/canary/canary.go

```diff
@@ -0,0 +1,74 @@
+package canary
```

Comment:
watchdog is another name to consider if you're looking for alternatives

Reply:
Yeah I considered that, but I always think of a watchdog timer where you have to call reset on it or it'll restart your thing. This is sort of the opposite of that in a way?